### ISSUE TITLE
Feature: scheme launch to app

### ISSUE THREAD

**Issue by crutkas** [Microsoft/WindowsTemplateStudio#506]
This will allow an application to to define a custom scheme URI and allow for a custom launch.  It should have stubbed area for parsing the query string in the area where it is handled.

Example would be `Contoso:?Addresses={address}&Body={body}`

https://docs.microsoft.com/en-us/windows/uwp/launch-resume/handle-uri-activation
https://docs.microsoft.com/en-us/windows/uwp/launch-resume/launch-default-app 

This would be an extension of #180.

**Comment by mrlacey** [comment]
Isn't this how a Cortana Skill (#145) would do this?

**Comment by mrlacey** [comment]
Also, this will need the ability to capture some extra information in the wizard. Just like as discussed for an enhancement (#325) or adding file headers (#477)

**Comment by crutkas** [comment]
Actually, yup, #145 would be how we'd pull that off

**Comment by mrlacey** [comment]
Having thought about how to implement this, I'm looking for feedback on plans (if seem right, not how to do this) to implement the following:

- The value for the protocol will be asked for in the wizard in the same way the name of a page can be specified. (New functionality for the validation rules and failure error message for this will be added to the template config)
- This feature will add another page (called ProtocolExample) to show passing information from a URI to a page within the app. This page is purely for demonstration and will include comments saying that it is for sample purposes only and should be removed
- For NavPanel and Tabs&Pivot projects, the activation handler will navigate to the example page within the shell/tab pages and have comments for how to navigate to a page, not within the shell
- The example page will display parameters passed to it (if any)
- Comments in the top of the activation handler will list supported protocol content formats
- The activation handler will show a basic example of querying the arguments for valid content in the CanHandleInternal method
- The current activation document will be extended with more information about protocol launching

**Comment by touseefbsb** [comment]
@mrlacey  the way you described seems reasonable and perfect :)

**Comment by mrlacey** [comment]
This is going to require a change to the wizard. The change will mean the new template won't work with the old wizard. Do we have a way of ensuring that a template can only be used with a certain version (or above) of the wizard? Or is there another way to stop anyone who doesn't update the VSIX have issues once the new template is public?
 //cc @ralarcon  @crutkas

**Comment by mrlacey** [comment]
bumping to 1.3 due to lack of time to integrate wizard change before 1.2 is due

**Comment by crutkas** [comment]
agreed

**Comment by ralarcon** [comment]
Sorry I missed this... we handle the compatibility thru the mayor.minor version numbers of the Wizard and the Templates content. No changes are required to be done now, just be sure that we increment (that by the way needs to be done anyway) the versions for templates content and wizard from 1.1.x.x to 1.2.x.x. in the build process.

The same for nightly and pre-release. Currently, the nightly release versions are in 0.14.x.x, when created, we will need to upgrade those too.  

In summary, we can deliver this for this version if is ready.

**Comment by mrlacey** [comment]
Will do without a logo asset being used but include a comment that a logo can be set.

**Comment by ralarcon** [comment]
Sounds good!

**Comment by mrlacey** [comment]
Updated docs now include details of the required image (thanks Tyler) so will include an example.

**Comment by mrlacey** [comment]
This is almost working but fails with the automated tests due to items not being renamed correctly.
Possibly related to this being due to being the first Feature which adds an actual page and that breaks something.
Possibly related to name collision detection.
I will continue to investigate.

**Comment by mrlacey** [comment]
sooooooooooooooo much manual testing needed for this !!!!!! :(

**Comment by mrlacey** [comment]
This took a disappointingly large amount of duplication across a total of ten templates (3 composition templates for each framework.) There's lots of duplication but this is the best balance I could find of things being manageable in terms of understanding what's going on and not needing any more special hacks to deal with the combinations of pieces of code required.
I'm open to refactoring suggestions if they bring clear benefits. Yes, there are definitely other ways of structuring this. Because this is done with a feature adding a page, existing composition templates for modifying pages don't work with this. :(

**Comment by mrlacey** [comment]
waiting on external review of this change from the Windows team

**Comment by mvegaca** [comment]
Checked on
Wizard: 0.14.17248.65001
Templates: 0.14.17247.1

**Comment by mvegaca** [comment]
This feature is not available from right click? Can it be added from right click @mrlacey?

**Comment by mrlacey** [comment]
Not enabled for right-click as it hasn't been tested for it.
Will test when I have time

**Comment by mvegaca** [comment]
We are near to close next version. Can I help you to review that?

**Comment by mvegaca** [comment]
I was reviewing thos files
- [Feature.UriSchemeExamplePage_Activation_Blank_SplitView](https://github.com/Microsoft/WindowsTemplateStudio/blob/dev/templates/_composition/MVVMBasic/Feature.UriSchemeExamplePage_Activation_Blank_SplitView/Activation/SchemeActivationHandler_postaction.cs)
- [Feature.UriSchemeExamplePage_Activation_TabbedPivot](https://github.com/Microsoft/WindowsTemplateStudio/blob/dev/templates/_composition/MVVMBasic/Feature.UriSchemeExamplePage_Activation_TabbedPivot/Activation/SchemeActivationHandler_postaction.cs)

All code added by composition templates are the same unless final NavigationService.Navigate.
I think that all method code could be added to [Feature.UriSchemeExamplePage](https://github.com/Microsoft/WindowsTemplateStudio/tree/dev/templates/_composition/MVVMBasic/Feature.UriSchemeExamplePage) composition template and compose only this different navigation line of code.

What do you think @mrlacey?

**Comment by crutkas** [comment]
@mvegaca, I would suggest doing a new issue for that optimization.  I want this one closed out.

**Comment by mrlacey** [comment]
Have created a new issue (#1095) to progress the suggestion so this can be closed out.
